File: blogServer/src/main/java/com/bolg/blogServer/service/auth/AuthService.java
package com.bolg.blogServer.service.auth;

import com.bolg.blogServer.dto.SignupRequest;
import com.bolg.blogServer.dto.UserDTO;

public interface AuthService {

     UserDTO createUser(SignupRequest signupRequest) throws Exception;

     Boolean hasUserWithEmail(String email);

     void forgotPassword(String email);

     void resetPassword(String token, String newPassword);

}

File: blogWeb/src/app/auth/services/auth/auth.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/";

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  constructor(private http: HttpClient) { }

  register(sigunupRequest: any): Observable<any> {
    return this.http.post(BASIC_URL + "auth/signup", sigunupRequest);
  };

  login(loginRequest: any): Observable<any> {
    return this.http.post<[]>(BASIC_URL + "auth/login", loginRequest);
  };

  forgotPassword(email: string): Observable<string> {
    return this.http.post(BASIC_URL + "auth/forgot-password", { email }, { responseType: 'text' });
  }

  resetPassword(token: string, newPassword: string): Observable<string> {
    return this.http.post(BASIC_URL + "auth/reset-password", { token, newPassword }, { responseType: 'text' });
  }

}

